cmake/externals/sentencepieceproject.cmake

```diff
@@ -1,20 +1,63 @@
 # spm is abbreviation of sentencepiece to meet the path length limits on Windows
+if(NOT _ONNXRUNTIME_EMBEDDED)
+  # If extensions wasn't built in ORT, we create fetchcontent the same 3rd party library as ORT
+  # So extensions is always consitent on the 3rd party libraries whether its build in ORT or not
 
-set(spm_patches)
-if(_ONNXRUNTIME_EMBEDDED)
-  # use protobuf provided by ORT
-  list(APPEND spm_patches "${PROJECT_SOURCE_DIR}/cmake/externals/sentencepieceproject.protobuf.patch")
-endif()
-if(IOS)
-  # fix for iOS build
-  list(APPEND spm_patches "${PROJECT_SOURCE_DIR}/cmake/externals/sentencepieceproject.ios.patch")
+  # TOOD: migrate to external abseil library
+  # include(abseil-cpp)
+  message(STATUS "Fetch protobuf")
+  FetchContent_Declare(
+    protobuf
+    GIT_REPOSITORY https://github.com/protocolbuffers/protobuf.git
+    GIT_TAG v3.20.2
+    PATCH_COMMAND git checkout . && git apply --ignore-space-change --ignore-whitespace ${PROJECT_SOURCE_DIR}/cmake/externals/protobuf_cmake.patch
+  )
+  set(protobuf_BUILD_TESTS OFF CACHE BOOL "Build tests")
+  set(protobuf_WITH_ZLIB OFF CACHE BOOL "Use zlib")
+
+  if(CMAKE_SYSTEM_NAME STREQUAL "Android")
+    set(protobuf_BUILD_PROTOC_BINARIES OFF CACHE BOOL "")
+  endif()
+  set(protobuf_BUILD_SHARED_LIBS OFF CACHE BOOL "")
+  set(protobuf_MSVC_STATIC_RUNTIME OFF CACHE BOOL "")
+  set(protobuf_DISABLE_RTTI ON CACHE BOOL "Disable RTTI")
+  FetchContent_GetProperties(protobuf)
+  if(NOT protobuf_POPULATED)
+    FetchContent_Populate(protobuf)
+    add_subdirectory(${protobuf_SOURCE_DIR}/cmake ${protobuf_BINARY_DIR} EXCLUDE_FROM_ALL)  
+  endif()
+
+  FetchContent_MakeAvailable(protobuf)
+  set_target_properties(libprotobuf PROPERTIES
+    FOLDER externals/google/protobuf)
+  set_target_properties(libprotobuf-lite PROPERTIES
+    FOLDER externals/google/protobuf)
+  if(NOT CMAKE_SYSTEM_NAME STREQUAL "Android")
+    set_target_properties(libprotoc PROPERTIES
+      FOLDER externals/google/protobuf)
+    set_target_properties(protoc PROPERTIES
+      FOLDER externals/google/protobuf)
+  endif()
 endif()
 
+# To avoid create a complicated logic to build protoc, especailly for mobile platforms, we use the pre-generated pb files
```

Comment:
```suggestion
# To avoid creating complicated logic to build protoc, especially for mobile platforms, we use the pre-generated pb files
```

Reply:
Something like https://marketplace.visualstudio.com/items?itemName=EWoodruff.VisualStudioSpellCheckerVS2022andLater is very handy for highlighting spelling errors in comments.